src/freenas/usr/local/libexec/nas/service_monitor.py

```diff
@@ -0,0 +1,28 @@
+#!/usr/local/bin/python
+import argparse
+from middlewared.client import Client
+from middlewared.client.utils import Struct
+
+
+class RCDServiceMonitoring(object):
+
+    def ad_monitor(self, client, option):
+        ad = Struct(client.call('datastore.query', 'directoryservice.activedirectory', None, {'get': True}))
+        if ad.ad_enable_monitor and ad.ad_enable:
+            if option == 'start':
+                return client.call('service.enable_test_service_connection', ad.ad_monitor_frequency, ad.ad_recover_retry, ad.ad_domainname, 3268, 'activedirectory')
+            elif option == 'stop':
+                return client.call('service.disable_test_service_connection', ad.ad_monitor_frequency, ad.ad_recover_retry, ad.ad_domainname, 3268, 'activedirectory')
+
+
+def main(option):
+    client = Client()
+    rcd = RCDServiceMonitoring()
+    rcd.ad_monitor(client, option)
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument("option", choices=['start', 'stop'], help="Control middlewared monitor services")
+
+    args = parser.parse_args()
+    main(args.option)
```

Comment:
My main concern is scattered logic between several places which might make it more complex/difficult to understand. I have made that mistake in several places which I am now trying to avoid.
If you think that will not be an issue, thats fine.